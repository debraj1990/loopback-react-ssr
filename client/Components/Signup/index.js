import React, { Component } from 'react';
import Form from '../Form';



class Signup extends Component{
    
    state = {
        email:'',
        password:'',
        name:''
    }

    onChange = ({target}) => {
        this.setState({[target.name]:target.value})
    }
    render(){
        const {name, email, password} = this.state
        return (
            <div>
                <Form
                  name={name}
                  email={email}
                  password={password}
                  onChange={this.onChange}
                  type='Signup'
                />
            </div>
        )
    }
}



export default Signup