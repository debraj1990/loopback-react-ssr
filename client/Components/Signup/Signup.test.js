import React from 'react';
import {configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';


import Signup from './'


configure({adapter: new Adapter });


describe('Login Component', () => {
  
  it('component should render the compoent without crushing', () => {
    const component = shallow(<Signup/>)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  });

  
  it('should render the forms', () => {
    const wrapper = shallow(<Signup error={false} />)
    expect(wrapper.find('Form')).toHaveLength(1)
  });

    
})