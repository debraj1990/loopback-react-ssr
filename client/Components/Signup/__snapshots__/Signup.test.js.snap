// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component component should render the compoent without crushing 1`] = `
<div>
  <h3
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    Hello Signup 
  </h3>
  <Form
    email=""
    name=""
    onChange={[Function]}
    onSubmit={[Function]}
    password=""
    type="Signup"
  />
</div>
`;
